### PR TITLE
Move router and metrics to api package

### LINKED ISSUE #206: Make end-to-end test use router
Move router into pkg/api from main. Makes sure our router is set up correctly.

### PR DESCRIPTION
**Pull request by antekresic**
Moving the router and metrics to separate package outside of main package
enables us to reuse it for our end-to-end tests and use the main router for
testing.

Closes #206

### CODE REVIEW

File: pkg/api/metrics.go
Comment (cevian) [inline pkg/api/metrics.go:169]: Optional additional refactor: I really think the code would look a lot better if this was in pkg/pgclient/metrics.go and looked like the existing pkg/pgmodel/metrics.go

There is no reason this needs to be part of of the metrics struct we pass around we can just make this local to pkg/pgclient

File: pkg/api/metrics.go
Comment (cevian) [inline pkg/api/metrics.go:18]: nit and optional refactor : do we need these vars? can't we just init the struct fields in `InitMetrics` and then register the fields? i.e.

```
InitMetrics() {
...
metrics = &Metrics{
          receivedSamples: prometheus.NewCounter(
		prometheus.CounterOpts{
			Namespace: util.PromNamespace,
			Name:      "received_samples_total",
			Help:      "Total number of received samples.",
		},
	)
...
prometheus.MustRegister(metrics.receivedSamples)
...
}
```

**Review by cevian — APPROVED** [review]
This looks great and cleans up a lot of code. Added 2 optional refactors you could do on top of this to make the code even cleaner. But I'll leave it up to you if you think worthwhile.

File: pkg/api/metrics.go
Comment (Harkishen-Singh) [inline pkg/api/metrics.go:138]: `MustRegister()` takes on args of metrics. Instead of calling on separate lines repeatedly, we should pass all these in just one MustRegister. This will lessen the complexity of stack trace, where multiple function calls are a overheads.

File: pkg/api/metrics.go
Comment (Harkishen-Singh) [inline pkg/api/metrics.go:205]: Sames goes over here as well.

**Review by Harkishen-Singh — CHANGES_REQUESTED** [review]

File: pkg/api/metrics.go
Comment (Harkishen-Singh) [reply pkg/api/metrics.go:138]: Like this:

https://github.com/prometheus/prometheus/blob/a282d2509971174301408c5a5f96946c478dfb0f/tsdb/db.go#L266

Comment (antekresic) [reply pkg/api/metrics.go]: Thanks for the tip, fixed!

File: pkg/api/metrics.go
Comment (antekresic) [reply pkg/api/metrics.go]: Good catch!

File: pkg/api/metrics.go
Comment (antekresic) [reply pkg/api/metrics.go]: Makes sense, will do it.

**Comment by antekresic** [comment]
@cevian implemented the refactoring you recommended. Could you take another look before merging?

File: cmd/timescale-prometheus/main.go
Comment (cevian) [inline cmd/timescale-prometheus/main.go:119]: Nit: can we put this call inside of pgclient.NewClient?

**Review by cevian — APPROVED** [review]
One nit but otherwise LGTM

File: pkg/pgmodel/end_to_end_tests/promql_write_endpoint_test.go
Comment (Harkishen-Singh) [inline pkg/pgmodel/end_to_end_tests/promql_write_endpoint_test.go:189]: Not sure, but isn't `t.Fatalf()` automatically exit?

**Comment by Harkishen-Singh** [comment]
Looks good. Didn't give an approval since not completely familiar with some parts. But, Mat and Josh have approved, so it should be good to go ahead.

**Pull request merged**
